source/service/k3s.rst

```diff
@@ -0,0 +1,952 @@
+
+After you are done, click on the button **Instantiate**. A virtual machine with the running service should be ready in a few minutes.
+
+.. include:: shared/report.txt
+
+.. _report_file_example:
+
+.. code::
+
+       ___   _ __    ___
+      / _ \ | '_ \  / _ \   OpenNebula Service Appliance
+     | (_) || | | ||  __/
+      \___/ |_| |_| \___|
+
+    All set and ready to serve 8)
+
+    onek3s-ip-192-168-122-10:~# cat /etc/one-appliance/config
+    [K3s]
+    k8s_connection = https://192.168.122.10:6443
+    k8s_master = 192.168.122.10
+    k8s_token = K107e18223bb5472f7dcf058a3b217c98583453ee3f90150504e2dfc9016bdbac71::server:78e26029234b950c88bf2361845b3568
+
+.. _k3s_context_param:
+
+Contextualization
+=================
+
+Contextualization parameters provided in the Virtual Machine template controls the initial VM configuration. Except for the `common set <https://docs.opennebula.io/stable/management_and_operations/references/template.html#context-section>`_ of parameters supported by every appliance on the OpenNebula Marketplace, there are few specific to the particular service appliance. The parameters should be provided in the ``CONTEXT`` section of the Virtual Machine template, read the OpenNebula `Management and Operations Guide <https://docs.opennebula.io/stable/management_and_operations/references/kvm_contextualization.html#set-up-the-virtual-machine-template>`__ for more details.
+
+====================================== ========= ==================== ========= ======== ===========
+Parameter                              Mandatory Default              Stage     Role     Description
+====================================== ========= ==================== ========= ======== ===========
+``ONEAPP_K8S_ADDRESS``                           routable IP          configure all      Master node address
+``ONEAPP_K8S_TOKEN``                                                  configure worker   Secret token - to join worker node to the cluster
+``ONEAPP_K8S_OVERRIDE_VERSION``                                       configure master   Custom K3s version (**it overrides the preinstalled K3s version**)
+``ONEAPP_K8S_NODENAME``                          hostname             configure master   Master Node Name
+``ONEAPP_K8S_PORT``                              ``6443``             configure master   K3s API port on which nodes communicate
+``ONEAPP_K8S_TAINTED_MASTER``                    ``no``               configure master   Master node acts as control-plane only (**you will need to add worker nodes**)
+``ONEAPP_K8S_CNI``                               ``flannel``          configure master   Container Network Interface (options: ``calico``, ``canal``)
+``ONEAPP_K8S_FLANNEL_BACKEND``                   ``vxlan``            configure master   Flannel CNI backend
```

Comment:
other options for flannel backed should be listed: ipsec, host-gw. Is wireguard also supported?

Reply:
Below in the description for the master node is a link to these other values - I did not want to list them because a) I did not test them b) they are not mine and can change in the future (new coming, old going) - wireguard needs probably extra packages/setup for it to work - not supported